worker/codalabworker/rest_client.py

```diff
@@ -51,8 +51,13 @@ def _make_request(self, method, path, query_params=None, headers=None,
         headers['X-Requested-With'] = 'XMLHttpRequest'
         if query_params is not None:
             path = path + '?' + urllib.urlencode(query_params)
-        request = urllib2.Request(self._base_url + path, data=data,
-                                  headers=headers)
+        # Everything needs to be utf-8 encoded or urllib2 will complain
+        if 'Content-Type' in headers:
+            headers['Content-Type'] += '; charset=utf-8'
+        if data and isinstance(data, unicode):
+            data = data.encode('utf-8')
+        request_url = (self._base_url + path).encode('utf-8')
```

Comment:
Yes for some reason, mismatching types between the url and the request body results in Unicode decode errors..